core/src/main/java/ar/com/wolox/wolmo/core/util/ImageProvider.kt

```diff
@@ -0,0 +1,247 @@
+        if (i.resolveActivity(context.packageManager) != null) {
+            fragment.startActivityForResult(i, requestCode)
+        } else {
+            toastFactory.show(errorResId)
+
+        }
+    }
+
+    /**
+     * Adds a given picture to the device images gallery.
+     *
+     * @param imageUri The [Uri] of the image
+     */
+    fun addPictureToDeviceGallery(imageUri: Uri) {
+        val mediaScanIntent = Intent(Intent.ACTION_MEDIA_SCANNER_SCAN_FILE)
+        mediaScanIntent.data = imageUri
+        context.sendBroadcast(mediaScanIntent)
+    }
+
+    /**
+     * Triggers an intent to go to the device's camera app, stores the image as 'filename'.'format',
+     * and returns its [Uri].
+     *
+     *
+     *
+     *
+     * Override the onActivityResult method in your fragment and specify behaviour
+     * for the provided request code. This method returns a File object that
+     * contains the picture taken. You can get the returned image as an [Uri] using
+     * [Uri.fromFile].
+     *
+     * @param fragment    A fragment where the get image intent is going to be called
+     * @param requestCode A request code for the intent
+     * @param filename    Filename for the future stored image
+     * @param format      Format (extension) for the future image
+     * @param errorResId  Resource id of the error string
+     * @return [Uri] of the newly stored image
+     */
+    fun getImageFromCamera(fragment: Fragment, requestCode: Int, filename: String,
+                           @ImageFormat format: String, @StringRes errorResId: Int): File? {
+
+        val i = Intent(MediaStore.ACTION_IMAGE_CAPTURE)
+
+        // Ensure that there's a camera app to handle the intent
+        if (i.resolveActivity(context.packageManager) == null) {
+            toastFactory.show(errorResId)
+            return null
+        }
+
+        val photoFile: File?
+        try {
+            photoFile = wolmoFileProvider.createTempFile(filename, format, Environment.DIRECTORY_DCIM)
+        } catch (ex: IOException) {
+            toastFactory.show(errorResId)
+            return null
+        }
+
+
+        val photoFileUri = wolmoFileProvider.getUriForFile(photoFile)
+
+        i.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
+        i.putExtra(MediaStore.EXTRA_OUTPUT, photoFileUri)
+        fragment.startActivityForResult(i, requestCode)
+
+
+        return photoFile
+    }
+
+    /**
+     * Get [byte[]] from an image file, represented by its [Uri].
+     *
+     * @param imageFileUri target image file URI
+     * @param format       image compress format
+     * @param quality      compress quality, between 0 and 100
+     * @param maxWidth     max width of the target image
+     * @param maxHeight    max height of the target image
+     * @return byte array with the formatted information of the image file, if the image exceeded
+     * boundaries, it's re scaled.
+     */
+    fun getImageAsByteArray(
```

Comment:
Same as some comments above. You could avoid return type and make it a one line function.

Reply:
ok :+1: